controller/array_action/array_mediator_svc.py

```diff
@@ -148,30 +164,65 @@ def _generate_volume_response(self, cli_volume):
             cli_volume.mdisk_grp_name,
             self.array_type)
 
-    def get_volume(self, volume_name, volume_context=None, volume_prefix=""):
-        logger.debug("Get volume : {}".format(volume_name))
-        cli_volume = None
+    def _generate_snapshot_response(self, cli_snapshot, source_volume_name):
+        return Snapshot(int(cli_snapshot.capacity),
+                        cli_snapshot.vdisk_UID,
+                        cli_snapshot.name,
+                        self.endpoint,
+                        source_volume_name,
+                        is_ready=True,
+                        array_type=self.array_type)
+
+    def _get_client_list_method(self, cli_object_type):
+        svcinfo = self.client.svcinfo
+        if cli_object_type == CLI_OBJECT_TYPE_VOLUME:
+            return lambda obj_id: svcinfo.lsvdisk(bytes=True, object_id=obj_id)
+        if cli_object_type == CLI_OBJECT_TYPE_FCMAP:
+            return lambda obj_id: svcinfo.lsfcmap(object_id=obj_id)
+        raise ValueError("No list method for cli object type: {}".format(cli_object_type))
+
+    def _get_cli_object(self, object_name_or_id, cli_object_type, not_exist_err=True):
```

Comment:
In these methods never raise exception if not exists. Just return None in such case as done with snapshots.

Reply:
I think the common usage actually does require raising an exception.
Could you elaborate on how should the mediator caller functions change accordingly?
(returning None all the way to the controller server, in most existing cases, is not really an option)

Reply:
i see only two relevant methods -  get_volume and get_snapshot
get_snapshot already returns None. So only get_volume is need to change. I opened ticket for it which should e very small (I checked. almost no changes). Instead of 4 methods u added we need only 2 - one for vol one for snapsshot.

Reply:
I don't think these two are the only relevant (public) methods.
I'm not sure I understood said ticket content - could you please add its number?
do you mean that this conversation should be resolved in said ticket?

Reply:
https://jira.xiv.ibm.com/browse/CSI-1446

Reply:
please remove internal links.
if you mean that this conversation should be resolved in CSI-1446, then please update its description, and we'll leave it open.
if not, then please clarify how would the other methods should be changed accordingly (e.g. delete)

Reply:
I mean this conversation depends on the ticket. So we leave the conversation open.

Reply:
This conversation can be closed after the ticket. Please open complementary ticket.

Reply:
fixes in https://github.com/IBM/ibm-block-csi-driver/pull/190